File: test/translations/classes/class_implements_interface.ts
interface IResourceWithPolicy {
  readonly env: string;
  addToPolicy(statement: string): void;
}

class MyResource implements IResourceWithPolicy {
  public readonly env: string;

  constructor(env: string) {
    this.env = env;
  }

  public addToPolicy(statement: string): void {
    console.log(statement);
  }
}

const r = new MyResource('production');
console.log(r.env);

File: test/translations/classes/class_implements_interface.cs
interface IResourceWithPolicy
{
    string Env { get; }
    void AddToPolicy(string statement);
}

class MyResource : IResourceWithPolicy
{
    public string Env { get; }

    public MyResource(string env)
    {
        Env = env;
    }

    public void AddToPolicy(string statement)
    {
        Console.WriteLine(statement);
    }
}

var r = new MyResource("production");
Console.WriteLine(r.Env);
